### PR TITLE
Enable FastJarFileSystem for faster builds and completions

### PR DESCRIPTION
**Pull request by PranavPurwar**

### CODE REVIEW

**Comment by tyron12233** [comment]
Thanks.

It would be better if you keep the original kotlin embeddable jar as gradle would replace the changed files automatically when packaging.

**Comment by PranavPurwar** [comment]
done

**Comment by El-o-heka** [comment]
Yeah done ✅

File: build-tools/kotlinc/src/main/java/org/jetbrains/kotlin/cli/jvm/compiler/jarfs/ZipImplementation.kt [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+fun MappedByteBuffer.parseCentralDirectory(): List<ZipEntryDescription> {
+    order(ByteOrder.LITTLE_ENDIAN)
+
+    var endOfCentralDirectoryOffset = capacity() - END_OF_CENTRAL_DIR_SIZE
+    while (endOfCentralDirectoryOffset >= 0) {
+        // header of "End of central directory"
+        if (getInt(endOfCentralDirectoryOffset) == 0x06054b50) break
+        endOfCentralDirectoryOffset--
+    }
+
```
Comment (devbrett25-collab) [inline build-tools/kotlinc/src/main/java/org/jetbrains/kotlin/cli/jvm/compiler/jarfs/ZipImplementation.kt]: https://github.com/tyron12233/CodeAssist/actions/workflows/android.ym

**Pull request merged**